From 1732caa80ca316e3298a51bd21beb924afcfb80c Mon Sep 17 00:00:00 2001
From: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
Date: Thu, 17 Oct 2024 11:30:54 +0800
Subject: [PATCH 1/2] configs: imx95: add config for RTE

Assigned the LPUART3 and related pins to A core NS SCMI agent.
Added 2 more SCMI agents with the same resource assignment
for A core NS env.

Signed-off-by: Hou Zhiqiang <Zhiqiang.Hou@nxp.com>
Upstream-Status: Pending
---
 configs/other/mx95rte.cfg | 984 ++++++++++++++++++++++++++++++++++++++
 1 file changed, 984 insertions(+)
 create mode 100755 configs/other/mx95rte.cfg

diff --git a/configs/other/mx95rte.cfg b/configs/other/mx95rte.cfg
new file mode 100755
index 0000000..82b99f1
--- /dev/null
+++ b/configs/other/mx95rte.cfg
@@ -0,0 +1,984 @@
+## ###################################################################
+##
+## Copyright 2023-2024 NXP
+##
+## Redistribution and use in source and binary forms, with or without modification,
+## are permitted provided that the following conditions are met:
+##
+## o Redistributions of source code must retain the above copyright notice, this list
+##   of conditions and the following disclaimer.
+##
+## o Redistributions in binary form must reproduce the above copyright notice, this
+##   list of conditions and the following disclaimer in the documentation and/or
+##   other materials provided with the distribution.
+##
+## o Neither the name of the copyright holder nor the names of its
+##   contributors may be used to endorse or promote products derived from this
+##   software without specific prior written permission.
+##
+## THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+## ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+## WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+## DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR
+## ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+## (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+## LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON
+## ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+## (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+## SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+##
+##
+## ###################################################################
+
+# SM configuration file for the Real-time-edge cases on MX95 EVK
+
+MAKE    soc=MIMX95, board=mcimx95evk, build=gcc_cross
+DOX     name=MX95RTE, desc="i.MX95 EVK Real-time-edge Configuration Data"
+
+include ../../devices/MIMX95/configtool/device.cfg
+
+#==========================================================================#
+# Board                                                                    #
+#==========================================================================#
+
+DEBUG_UART_INSTANCE  2
+DEBUG_UART_BAUDRATE  115200
+
+PMIC_I2C_INSTANCE    1
+PMIC_I2C_BAUDRATE    400000
+
+#==========================================================================#
+# ELE Domain                                                               #
+#==========================================================================#
+
+DOM0                did=0
+
+OWNER:              perm=rw
+
+# Resources
+
+# Memory
+
+DDR                 OWNER, begin=0x080000000, end=0xFFFFFFFFF
+M33_TCM_SYS         OWNER, begin=0x20200000, size=256K
+OCRAM               OWNER, begin=0x20480000, size=352K
+
+#==========================================================================#
+# ISP Domain                                                               #
+#==========================================================================#
+
+DOM10               did=10
+
+OWNER:              perm=rw
+
+#==========================================================================#
+# V2X Domain                                                               #
+#==========================================================================#
+
+DOM12               did=12
+
+DFMT1:              sa=bypass, pa=bypass
+
+V2X_FH
+
+#==========================================================================#
+# SM M33 EENV                                                              #
+#==========================================================================#
+
+LM0                 name="SM", rpc=none, boot=1, did=2, safe=feenv
+
+DFMT0:              sa=secure
+DFMT1:              sa=secure, pa=privileged
+OWNER:              perm=sec_rw, api=all
+ACCESS:             perm=sec_rw, api=all, mdid=none
+READONLY:           perm=ro
+TEST_MU:            perm=sec_rw
+
+EXEC:               perm=sec_rwx
+DATA:               perm=sec_rw
+
+MODE                msel=1, boot=1
+MODE                msel=2, boot=1
+
+# API
+
+CLK_BUSAON               OWNER
+CLK_BUSM7                OWNER
+CLK_BUSNETCMIX           OWNER
+CLK_BUSWAKEUP            OWNER
+CLK_CAMAPB               OWNER
+CLK_CAMAXI               OWNER
+CLK_CAMISI               OWNER
+CLK_DISPAPB              OWNER
+CLK_DISPAXI              OWNER
+CLK_ELE                  OWNER
+CLK_FRO                  OWNER
+CLK_GPU                  OWNER
+CLK_GPUAPB               OWNER
+CLK_HSIO                 OWNER
+CLK_NOC                  OWNER
+CLK_NOCAPB               OWNER
+CLK_NPU                  OWNER
+CLK_NPUAPB               OWNER
+CLK_OSC24M               OWNER
+CLK_OSC32K               OWNER
+CLK_SYSPLL1_PFD0         OWNER
+CLK_SYSPLL1_PFD0_DIV2    OWNER
+CLK_SYSPLL1_PFD0_UNGATED OWNER
+CLK_SYSPLL1_PFD1         OWNER
+CLK_SYSPLL1_PFD1_DIV2    OWNER
+CLK_SYSPLL1_PFD1_UNGATED OWNER
+CLK_SYSPLL1_PFD2         OWNER
+CLK_SYSPLL1_PFD2_DIV2    OWNER
+CLK_SYSPLL1_PFD2_UNGATED OWNER
+CLK_SYSPLL1_VCO          OWNER
+CLK_TEMPSENSE_GPR_SEL    OWNER
+CLK_TMU                  OWNER
+CLK_VPU                  OWNER
+CLK_VPUAPB               OWNER
+CLK_VPUJPEG              OWNER
+CLK_WAKEUPAXI            OWNER
+
+# Resources
+
+M33P                OWNER  # CPUs must be first
+ANATOP              OWNER
+ATU_A               OWNER
+ATU_M               OWNER
+AXBS_AON            OWNER
+BBNSM               OWNER
+BLK_CTRL_BBSMMIX    OWNER
+BLK_CTRL_DDRMIX     OWNER
+BLK_CTRL_NOCMIX     OWNER
+BLK_CTRL_NS_AONMIX  OWNER
+BLK_CTRL_S_AONMIX   OWNER
+BLK_CTRL_WAKEUPMIX  OWNER
+CCM                 OWNER
+DAP                 OWNER
+DDR_CTRL            OWNER
+DDR_PHY             OWNER
+DDR_PM              OWNER
+FSB                 READONLY
+GIC                 ACCESS
+GPC                 OWNER
+GPIO1               OWNER
+GPR0                OWNER
+GPR1                OWNER
+GPR2                OWNER
+GPR3                OWNER
+GPV_CAMERA          OWNER
+GPV_CENTRAL         OWNER
+GPV_DISPLAY         OWNER
+GPV_HSIO            OWNER
+GPV_MAIN            OWNER
+GPV_MEGA            OWNER
+GPV_VPU             OWNER
+IOMUX_GPR           OWNER
+IOMUXC              OWNER
+JTAG                OWNER
+LPI2C1              OWNER
+LPUART2             OWNER
+M33_CACHE_CTRL      OWNER
+M33_PCF             OWNER
+M33_PSF             OWNER
+M33_TCM_ECC         OWNER
+MSGINTR1            OWNER
+MU1_A               TEST_MU
+MU1_B               OWNER
+MU2_A               TEST_MU
+MU2_B               OWNER
+MU3_A               TEST_MU
+MU3_B               OWNER
+MU4_A               TEST_MU
+MU4_B               OWNER
+MU5_A               TEST_MU
+MU5_B               OWNER
+MU6_A               TEST_MU
+MU6_B               OWNER
+MU_ELE0             OWNER
+ROMCP_M33           OWNER
+ELE                 OWNER
+SRAM_CTL_1          OWNER
+SRAM_CTL_N          OWNER
+SRC                 OWNER
+SYSCTR_CTL          OWNER
+SYSCTR_RD           OWNER
+TEMP_A55            OWNER
+TRDC_A              OWNER
+TRDC_C              OWNER
+TRDC_D              OWNER
+TRDC_E              OWNER
+TRDC_G              OWNER
+TRDC_H              OWNER
+TRDC_M              OWNER
+TRDC_N              OWNER
+TRDC_V              OWNER
+TRDC_W              OWNER
+TSTMR1              OWNER
+WDOG1               OWNER
+WDOG2               OWNER
+
+# SAF
+
+C_STCU              OWNER
+CMU_A1              OWNER
+CMU_A2              OWNER
+CMU_ANA             OWNER
+CMU_DDR1            OWNER
+CMU_DDR2            OWNER
+CMU_N1              OWNER
+CMU_N2              OWNER
+CMU_W1              OWNER
+CMU_W2              OWNER
+CRC_A               OWNER
+DMA_CRC             OWNER
+EIM_A               OWNER
+EIM_N               OWNER
+EIM_NPU             OWNER
+EIM_W               OWNER
+ERM_A               OWNER
+ERM_NPU             OWNER
+ERM_W               OWNER
+FCCU                OWNER
+FCCU_FHID           OWNER
+INTM                OWNER
+L_STCU_A            OWNER
+L_STCU_DDR          OWNER
+L_STCU_N            OWNER
+L_STCU_NPUMIX       OWNER
+
+# Pins
+
+PIN_FCCU_ERR0         OWNER
+PIN_I2C1_SCL          OWNER
+PIN_I2C1_SDA          OWNER
+PIN_PDM_BIT_STREAM1   OWNER
+PIN_UART2_RXD         OWNER
+PIN_UART2_TXD         OWNER
+PIN_WDOG_ANY          OWNER
+
+# Memory
+
+M33_TCM_CODE        EXEC, begin=0x201C0000, size=256K
+M33_TCM_SYS         EXEC, begin=0x20200000, size=256K
+M33_ROM             EXEC, begin=0x00000000, end=0x0003FFFF
+M7MIX               DATA, begin=0x4A000000, end=0x4A0AFFFF
+
+# Faults
+
+FAULT_SW3           OWNER, reaction=grp_reset
+FAULT_SW4           OWNER, reaction=sys_reset
+
+#==========================================================================#
+# M7 EENV                                                                  #
+#==========================================================================#
+
+LM1                 name="M7", rpc=scmi, boot=2, skip=1, did=4, safe=seenv
+
+DFMT0:              sa=secure
+DFMT1:              sa=secure, pa=privileged
+OWNER:              perm=rw, api=all
+READONLY:           perm=ro
+
+EXEC:               perm=full
+DATA:               perm=rw
+
+MODE                msel=1, boot=2
+
+# Start/Stop (mSel=0)
+
+PD_M7               stop=2
+CPU_M7P             start=1, stop=1
+
+# Start/Stop (mSel=1)
+
+PD_M7               msel=2, stop=2
+CPU_M7P             msel=1, start=1, stop=1
+
+# Start/Stop (mSel=2)
+
+PD_M7               msel=2, stop=2
+CPU_M7P             msel=2, start=1, stop=1
+
+# RPC Interface
+
+SCMI_AGENT0         name="M7"
+MAILBOX             type=mu, mu=9, test=8, priority=high
+CHANNEL             db=0, xport=smt, check=crc32, rpc=scmi, type=a2p, \
+                    test=default
+CHANNEL             db=1, xport=smt, check=crc32, rpc=scmi, \
+                    type=p2a_notify, notify=24
+CHANNEL             db=2, xport=smt, check=crc32, rpc=scmi, \
+                    type=p2a_priority
+
+# API
+
+BRD_SM_CTRL_BUTTON        api=notify
+BRD_SM_CTRL_PCA2131       api=all
+BRD_SM_CTRL_TEST          api=all
+BRD_SM_RTC_PCA2131        api=all
+BRD_SM_SENSOR_TEMP_PF09   api=set
+BRD_SM_SENSOR_TEMP_PF5301 api=set
+BRD_SM_SENSOR_TEMP_PF5302 api=set
+FUSA                      api=exclusive
+LMM_2                     api=all
+SENSOR_TEMP_ANA           api=all
+SYS                       api=all
+
+# Resources
+
+M7P                 OWNER  # CPUs must be first
+PD_M7               test
+BUTTON              api=notify
+CAN_FD1             OWNER
+FSB                 READONLY
+IRQSTEER_M7         OWNER
+LPIT1               OWNER
+LPTMR1              OWNER
+LPTMR2              OWNER
+LPTPM1              OWNER
+LPUART3             OWNER, test
+MSGINTR2            OWNER
+MU5_A               OWNER
+MU7_B               OWNER
+MU8_B               OWNER
+MU_ELE5             OWNER
+RTC                 api=priv
+SYSCTR_RD           READONLY
+TSTMR2              OWNER
+V2X_SHE1            OWNER
+WDOG5               OWNER
+
+# Pins
+
+PIN_GPIO_IO14       OWNER
+PIN_GPIO_IO15       OWNER
+
+# Memory
+
+M7MIX               DATA, begin=0x020380000, end=0x02047FFFF
+M7MIX               DATA, begin=0x04A000000, end=0x04A0AFFFF
+DDR                 EXEC, begin=0x080000000, end=0x089FFFFFF
+
+# Faults
+
+FAULT_WDOG5         OWNER, reaction=lm_reset
+FAULT_M7_LOCKUP     OWNER, reaction=lm_reset
+FAULT_M7_RESET      OWNER, reaction=lm_reset
+FAULT_SW0           OWNER, reaction=fusa
+FAULT_SW1           OWNER, reaction=lm_reset
+
+#==========================================================================#
+# A55 secure EENV                                                          #
+#==========================================================================#
+
+LM2                 name="AP", rpc=scmi, boot=3, skip=1, did=3, default
+
+DFMT0:              sa=bypass
+DFMT1:              sa=secure, pa=privileged
+OWNER:              perm=sec_rw, api=all
+CONTROL:            perm=none, api=all
+
+EXEC:               perm=sec_rwx
+DATA:               perm=rw
+
+# Start/Stop (mSel=0)
+
+VOLT_ARM            start=1|1, stop=9
+PD_A55P             start=2, stop=8, test
+PD_A55C0            stop=7
+PD_A55C1            stop=6
+PD_A55C2            stop=5
+PD_A55C3            stop=4
+PD_A55C4            stop=3
+PD_A55C5            stop=2
+PERF_A55            start=3|3
+CPU_A55C0           start=4
+CPU_A55P            stop=1
+
+# Start/Stop (mSel=1)
+
+VOLT_ARM            msel=1, start=1|1, stop=9
+PD_A55P             msel=1, start=2, stop=8
+PD_A55C0            msel=1, stop=7
+PD_A55C1            msel=1, stop=6
+PD_A55C2            msel=1, stop=5
+PD_A55C3            msel=1, stop=4
+PD_A55C4            msel=1, stop=3
+PD_A55C5            msel=1, stop=2
+PERF_A55            msel=1, start=3|3
+CPU_A55C0           msel=1, start=4
+CPU_A55P            msel=1, stop=1
+
+# Start/Stop (mSel=2)
+
+VOLT_ARM            msel=2, start=1|1, stop=9
+PD_A55P             msel=2, start=2, stop=8
+PD_A55C0            msel=2, stop=7
+PD_A55C1            msel=2, stop=6
+PD_A55C2            msel=2, stop=5
+PD_A55C3            msel=2, stop=4
+PD_A55C4            msel=2, stop=3
+PD_A55C5            msel=2, stop=2
+PERF_A55            msel=2, start=3|3
+CPU_A55C0           msel=2, start=4
+CPU_A55P            msel=2, stop=1
+
+# RPC Interface
+
+SCMI_AGENT1         name="AP-S", secure
+MAILBOX             type=mu, mu=1, test=0
+CHANNEL             db=0, xport=smt, rpc=scmi, type=a2p
+CHANNEL             db=1, xport=smt, rpc=scmi, type=p2a_notify, notify=24
+
+# API
+
+PERF_A55            CONTROL
+PERF_DRAM           CONTROL
+PERLPI_GPIO2        api=all
+PERLPI_GPIO3        api=all
+PERLPI_GPIO4        api=all
+PERLPI_GPIO5        api=all
+PERLPI_CAN2         api=all
+PERLPI_CAN3         api=all
+PERLPI_CAN4         api=all
+PERLPI_CAN5         api=all
+PERLPI_LPUART1      api=all
+PERLPI_LPUART4      api=all
+PERLPI_LPUART5      api=all
+PERLPI_LPUART6      api=all
+PERLPI_LPUART7      api=all
+PERLPI_LPUART8      api=all
+PERLPI_WDOG3        api=all
+PERLPI_WDOG4        api=all
+SYS                 api=all
+
+# Resources
+
+A55C0               OWNER  # CPUs must be first
+A55C1               OWNER  # CPUs must be first
+A55C2               OWNER  # CPUs must be first
+A55C3               OWNER  # CPUs must be first
+A55C4               OWNER  # CPUs must be first
+A55C5               OWNER  # CPUs must be first
+A55P                OWNER
+ARM_PLL             OWNER
+MU1_A               OWNER
+MU_ELE1             OWNER
+MU_ELE2             OWNER
+
+# Pins
+
+# Memory
+
+DDR                 EXEC, begin=0x08A000000, end=0x08DFFFFFF
+
+# Faults
+
+FAULT_WDOG3         OWNER, reaction=lm_reset
+FAULT_WDOG4         OWNER, reaction=lm_reset
+FAULT_SW2           OWNER, reaction=lm_reset
+
+#==========================================================================#
+# A55 non-secure EENV                                                      #
+#==========================================================================#
+
+DFMT0:              sa=nonsecure
+DFMT1:              sa=nonsecure, pa=privileged
+
+OWNER:              perm=rw, api=all
+CONTROL:            perm=rw, api=all, mdid=none
+READONLY:           perm=ro
+
+EXEC:               perm=full
+DATA:               perm=rw
+
+# RPC Interface of A55 NS agent0
+
+SCMI_AGENT2         name="AP-NS-agent0"
+MAILBOX             type=mu, mu=3, test=2
+CHANNEL             db=0, xport=smt, rpc=scmi, type=a2p
+CHANNEL             db=1, xport=smt, rpc=scmi, type=p2a_notify
+
+# API
+
+BRD_SM_CTRL_BT_WAKE       api=notify
+BRD_SM_CTRL_BUTTON        api=notify
+BRD_SM_CTRL_PCIE1_WAKE    api=notify
+BRD_SM_CTRL_PCIE2_WAKE    api=notify
+BRD_SM_CTRL_SD3_WAKE      api=notify
+BRD_SM_RTC_PCA2131        api=notify
+BRD_SM_SENSOR_TEMP_PF09   OWNER
+BRD_SM_SENSOR_TEMP_PF5301 api=set
+BRD_SM_SENSOR_TEMP_PF5302 api=set
+CLK_DISP1PIX              OWNER
+CLK_DISP2PIX              OWNER
+CLK_DISP3PIX              OWNER
+CLK_HSIOPCIEAUX           OWNER
+CLK_HSIOPLL               OWNER
+CLK_HSIOPLL_VCO           OWNER
+CLK_LDBPLL                OWNER
+CLK_LDBPLL_VCO            OWNER
+CLK_USBPHYBURUNIN         OWNER
+CLK_VPUDSP                OWNER
+LMM_1                     api=notify
+PERF_A55                  OWNER
+PERF_DRAM                 OWNER
+SENSOR_TEMP_A55           OWNER
+SENSOR_TEMP_ANA           api=set
+SYS                       api=notify
+
+# Resources
+
+ADC                 OWNER
+AUDIO_PLL1          OWNER
+AUDIO_PLL2          OWNER
+BLK_CTRL_CAMERAMIX  OWNER
+BLK_CTRL_DISPLAYMIX OWNER
+BLK_CTRL_GPUMIX     OWNER
+BLK_CTRL_HSIOMIX    OWNER
+BLK_CTRL_NETCMIX    OWNER
+BLK_CTRL_NPUMIX     OWNER
+BLK_CTRL_VPUMIX     OWNER
+BUTTON              OWNER, test
+CAMERA1             OWNER
+CAMERA2             OWNER
+CAMERA3             OWNER
+CAMERA4             OWNER
+CAMERA5             OWNER
+CAMERA6             OWNER
+CAMERA7             OWNER
+CAMERA8             OWNER
+CAN_FD2             OWNER
+CAN_FD3             OWNER
+CAN_FD4             OWNER
+CAN_FD5             OWNER
+CLOCK_EXT           OWNER
+CLOCK_EXT1          OWNER
+CLOCK_EXT2          OWNER
+CLOCK_OUT1          OWNER
+CLOCK_OUT2          OWNER
+CLOCK_OUT3          OWNER
+CLOCK_OUT4          OWNER
+DC                  OWNER, test
+DC0                 OWNER
+DC1                 OWNER
+DC_2DBLIT           OWNER
+DC_BLITINT          OWNER
+DC_CMDSEQ           OWNER
+DC_DISPENG          OWNER
+DC_DISPENG_INT      OWNER
+DC_FL0              OWNER
+DC_FL1              OWNER
+DC_INT_CTL          OWNER
+DC_PIXENGINE        OWNER
+DC_XPC              OWNER
+DC_YUV0             OWNER
+DC_YUV1             OWNER
+DC_YUV2             OWNER
+DC_YUV3             OWNER
+DDR_PM              OWNER
+EDMA1_MP            OWNER
+EDMA1_CH0           OWNER
+EDMA1_CH1           OWNER
+EDMA1_CH2           OWNER
+EDMA1_CH3           OWNER
+EDMA1_CH4           OWNER
+EDMA1_CH5           OWNER
+EDMA1_CH6           OWNER
+EDMA1_CH7           OWNER
+EDMA1_CH8           OWNER
+EDMA1_CH9           OWNER
+EDMA1_CH10          OWNER
+EDMA1_CH11          OWNER
+EDMA1_CH12          OWNER
+EDMA1_CH13          OWNER
+EDMA1_CH14          OWNER
+EDMA1_CH15          OWNER
+EDMA1_CH16          OWNER
+EDMA1_CH17          OWNER
+EDMA1_CH18          OWNER
+EDMA1_CH19          OWNER
+EDMA1_CH20          OWNER
+EDMA1_CH21          OWNER
+EDMA1_CH22          OWNER
+EDMA1_CH23          OWNER
+EDMA1_CH24          OWNER
+EDMA1_CH25          OWNER
+EDMA1_CH26          OWNER
+EDMA1_CH27          OWNER
+EDMA1_CH28          OWNER
+EDMA1_CH29          OWNER
+EDMA1_CH30          OWNER
+EDMA1_CH31          OWNER
+EDMA2_MP            OWNER
+EDMA2_CH0_1         OWNER
+EDMA2_CH2_3         OWNER
+EDMA2_CH4_5         OWNER
+EDMA2_CH6_7         OWNER
+EDMA2_CH8_9         OWNER
+EDMA2_CH10_11       OWNER
+EDMA2_CH12_13       OWNER
+EDMA2_CH14_15       OWNER
+EDMA2_CH16_17       OWNER
+EDMA2_CH18_19       OWNER
+EDMA2_CH20_21       OWNER
+EDMA2_CH22_23       OWNER
+EDMA2_CH24_25       OWNER
+EDMA2_CH26_27       OWNER
+EDMA2_CH28_29       OWNER
+EDMA2_CH30_31       OWNER
+EDMA2_CH32_33       OWNER
+EDMA2_CH34_35       OWNER
+EDMA2_CH36_37       OWNER
+EDMA2_CH38_39       OWNER
+EDMA2_CH40_41       OWNER
+EDMA2_CH42_43       OWNER
+EDMA2_CH44_45       OWNER
+EDMA2_CH46_47       OWNER
+EDMA2_CH48_49       OWNER
+EDMA2_CH50_51       OWNER
+EDMA2_CH52_53       OWNER
+EDMA2_CH54_55       OWNER
+EDMA2_CH56_57       OWNER
+EDMA2_CH58_59       OWNER
+EDMA2_CH60_61       OWNER
+EDMA2_CH62_63       OWNER
+EDMA3_MP            OWNER
+EDMA3_CH0_1         OWNER
+EDMA3_CH2_3         OWNER
+EDMA3_CH4_5         OWNER
+EDMA3_CH6_7         OWNER
+EDMA3_CH8_9         OWNER
+EDMA3_CH10_11       OWNER
+EDMA3_CH12_13       OWNER
+EDMA3_CH14_15       OWNER
+EDMA3_CH16_17       OWNER
+EDMA3_CH18_19       OWNER
+EDMA3_CH20_21       OWNER
+EDMA3_CH22_23       OWNER
+EDMA3_CH24_25       OWNER
+EDMA3_CH26_27       OWNER
+EDMA3_CH28_29       OWNER
+EDMA3_CH30_31       OWNER
+EDMA3_CH32_33       OWNER
+EDMA3_CH34_35       OWNER
+EDMA3_CH36_37       OWNER
+EDMA3_CH38_39       OWNER
+EDMA3_CH40_41       OWNER
+EDMA3_CH42_43       OWNER
+EDMA3_CH44_45       OWNER
+EDMA3_CH46_47       OWNER
+EDMA3_CH48_49       OWNER
+EDMA3_CH50_51       OWNER
+EDMA3_CH52_53       OWNER
+EDMA3_CH54_55       OWNER
+EDMA3_CH56_57       OWNER
+EDMA3_CH58_59       OWNER
+EDMA3_CH60_61       OWNER
+EDMA3_CH62_63       OWNER
+FLEXIO1             OWNER
+FLEXIO2             OWNER
+FLEXSPI1            OWNER
+FSB                 READONLY
+GIC                 OWNER
+GPIO2               OWNER
+GPIO3               OWNER
+GPIO4               OWNER
+GPIO5               OWNER
+GPR4                OWNER, test
+GPR5                OWNER
+GPR6                OWNER
+GPR7                OWNER
+GPU_NPROT           OWNER
+GPU_PROT            OWNER, test
+I3C1                OWNER
+I3C2                OWNER
+ISI1                OWNER
+ISI2                OWNER
+ISI3                OWNER
+ISI4                OWNER
+ISI5                OWNER
+ISI6                OWNER
+ISI7                OWNER
+ISI8                OWNER
+ISP_CPU             OWNER
+ISP_MGR             OWNER, test
+JPEG_DEC            OWNER
+LPI2C2              OWNER
+LPI2C3              OWNER
+LPI2C4              OWNER
+LPI2C5              OWNER
+LPI2C6              OWNER
+LPI2C7              OWNER
+LPI2C8              OWNER
+LPIT2               OWNER
+LPSPI1              OWNER
+LPSPI2              OWNER
+LPSPI3              OWNER
+LPSPI4              OWNER
+LPSPI5              OWNER
+LPSPI6              OWNER
+LPSPI7              OWNER
+LPSPI8              OWNER
+LPTPM2              OWNER
+LPTPM3              OWNER
+LPTPM4              OWNER
+LPTPM5              OWNER
+LPTPM6              OWNER
+LPUART1             OWNER
+LPUART3             CONTROL
+LPUART4             OWNER
+LPUART5             OWNER
+LPUART6             OWNER
+LPUART7             OWNER
+LPUART8             OWNER, test
+LVDS                OWNER
+MIPI_CSI0           OWNER
+MIPI_CSI1           OWNER
+MIPI_DSI            OWNER
+MIPI_PHY            OWNER
+MJPEG_DEC1          OWNER
+MJPEG_DEC2          OWNER
+MJPEG_DEC3          OWNER
+MJPEG_DEC4          OWNER
+MJPEG_ENC           OWNER
+MJPEG_ENC1          OWNER
+MJPEG_ENC2          OWNER
+MJPEG_ENC3          OWNER
+MJPEG_ENC4          OWNER
+MU2_A               OWNER
+MU3_A               OWNER
+MU4_A               OWNER
+MU6_A               OWNER
+MU7_A               OWNER
+MU8_A               OWNER
+MU_ELE3             OWNER
+MU_ELE4             OWNER
+NETC                OWNER, test
+NETC0               OWNER
+NETC1               OWNER
+NETC2               OWNER
+NETC_ECAM           OWNER
+NETC_EMDIO0         OWNER
+NETC_IERB           OWNER
+NETC_LDID1          OWNER, kpa=1, sid=0x20
+NETC_LDID2          OWNER, kpa=1, sid=0x21
+NETC_LDID3          OWNER, kpa=1, sid=0x22
+NETC_LDID4          OWNER, kpa=1, sid=0x23
+NETC_LDID5          OWNER, kpa=1, sid=0x24
+NETC_LDID6          OWNER, kpa=1, sid=0x25
+NETC_LDID7          OWNER, kpa=1, sid=0x26
+NETC_LDID8          OWNER, kpa=1, sid=0x27
+NETC_PRB            OWNER
+NETC_TIMER0         OWNER
+NETC_VSI0           OWNER
+NETC_VSI1           OWNER
+NETC_VSI2           OWNER
+NETC_VSI3           OWNER
+NETC_VSI4           OWNER
+NETC_VSI5           OWNER
+NPU                 OWNER, test
+PCI1_LUT0           OWNER, kpa=0, sid=0x10
+PCI1_LUT1           OWNER, kpa=0, sid=0x11
+PCI1_LUT2           OWNER, kpa=0, sid=0x12
+PCI1_LUT3           OWNER, kpa=0, sid=0x13
+PCI1_LUT4           OWNER, kpa=0, sid=0x14
+PCI1_LUT5           OWNER, kpa=0, sid=0x15
+PCI1_LUT6           OWNER, kpa=0, sid=0x16
+PCI1_LUT7           OWNER, kpa=0, sid=0x17
+PCI2_LUT0           OWNER, kpa=0, sid=0x18
+PCI2_LUT1           OWNER, kpa=0, sid=0x19
+PCI2_LUT2           OWNER, kpa=0, sid=0x1a
+PCI2_LUT3           OWNER, kpa=0, sid=0x1b
+PCI2_LUT4           OWNER, kpa=0, sid=0x1c
+PCI2_LUT5           OWNER, kpa=0, sid=0x1d
+PCI2_LUT6           OWNER, kpa=0, sid=0x1e
+PCI2_LUT7           OWNER, kpa=0, sid=0x1f
+PCIE1_OUT           OWNER
+PCIE1_ROOT          OWNER
+PCIE2_OUT           OWNER
+PCIE2_ROOT          OWNER
+PDM                 OWNER
+RTC                 OWNER, test
+SAI1                OWNER
+SAI2                OWNER
+SAI3                OWNER
+SAI4                OWNER
+SAI5                OWNER
+SEMA41              OWNER
+SEMA42              OWNER
+SMMU                OWNER
+SPDIF1              OWNER
+SYSCTR_CMP          OWNER
+SYSCTR_RD_STOP      READONLY
+USB1                OWNER, test, kpa=0, sid=0xe
+USB2                OWNER, kpa=0, sid=0xf
+USDHC1              OWNER
+USDHC2              OWNER
+USDHC3              OWNER
+V2X_APP0            OWNER
+V2X_DEBUG           OWNER
+V2X_FH              OWNER, mdid=none
+V2X_HSM1            OWNER
+V2X_HSM2            OWNER
+V2X_SHE0            OWNER
+VIDEO_PLL1          OWNER
+VPU                 OWNER, test
+VPU1                OWNER
+VPU2                OWNER
+VPU3                OWNER
+VPU4                OWNER
+WDOG3               OWNER
+WDOG4               OWNER
+XSPI                OWNER
+
+# Pins
+
+PIN_CCM_CLKO1        OWNER
+PIN_CCM_CLKO2        OWNER
+PIN_CCM_CLKO3        OWNER
+PIN_CCM_CLKO4        OWNER
+PIN_DAP_TCLK_SWCLK   OWNER
+PIN_DAP_TDI          OWNER
+PIN_DAP_TDO_TRACESWO OWNER
+PIN_DAP_TMS_SWDIO    OWNER
+PIN_ENET1_MDC        OWNER
+PIN_ENET1_MDIO       OWNER
+PIN_ENET1_RD0        OWNER
+PIN_ENET1_RD1        OWNER
+PIN_ENET1_RD2        OWNER
+PIN_ENET1_RD3        OWNER
+PIN_ENET1_RX_CTL     OWNER
+PIN_ENET1_RXC        OWNER
+PIN_ENET1_TD0        OWNER
+PIN_ENET1_TD1        OWNER
+PIN_ENET1_TD2        OWNER
+PIN_ENET1_TD3        OWNER
+PIN_ENET1_TX_CTL     OWNER
+PIN_ENET1_TXC        OWNER
+PIN_ENET2_MDC        OWNER
+PIN_ENET2_MDIO       OWNER
+PIN_ENET2_RD0        OWNER
+PIN_ENET2_RD1        OWNER
+PIN_ENET2_RD2        OWNER
+PIN_ENET2_RD3        OWNER
+PIN_ENET2_RX_CTL     OWNER
+PIN_ENET2_RXC        OWNER
+PIN_ENET2_TD0        OWNER
+PIN_ENET2_TD1        OWNER
+PIN_ENET2_TD2        OWNER
+PIN_ENET2_TD3        OWNER
+PIN_ENET2_TX_CTL     OWNER
+PIN_ENET2_TXC        OWNER
+PIN_GPIO_IO00        OWNER
+PIN_GPIO_IO01        OWNER
+PIN_GPIO_IO02        OWNER
+PIN_GPIO_IO03        OWNER
+PIN_GPIO_IO04        OWNER
+PIN_GPIO_IO05        OWNER
+PIN_GPIO_IO06        OWNER
+PIN_GPIO_IO07        OWNER
+PIN_GPIO_IO08        OWNER
+PIN_GPIO_IO09        OWNER
+PIN_GPIO_IO10        OWNER
+PIN_GPIO_IO11        OWNER
+PIN_GPIO_IO12        OWNER
+PIN_GPIO_IO13        OWNER
+PIN_GPIO_IO14        CONTROL
+PIN_GPIO_IO15        CONTROL
+PIN_GPIO_IO16        OWNER
+PIN_GPIO_IO17        OWNER
+PIN_GPIO_IO18        OWNER
+PIN_GPIO_IO19        OWNER
+PIN_GPIO_IO20        OWNER
+PIN_GPIO_IO21        OWNER
+PIN_GPIO_IO22        OWNER
+PIN_GPIO_IO23        OWNER
+PIN_GPIO_IO24        OWNER
+PIN_GPIO_IO25        OWNER
+PIN_GPIO_IO26        OWNER
+PIN_GPIO_IO27        OWNER
+PIN_GPIO_IO28        OWNER
+PIN_GPIO_IO29        OWNER
+PIN_GPIO_IO30        OWNER
+PIN_GPIO_IO31        OWNER
+PIN_GPIO_IO32        OWNER
+PIN_GPIO_IO33        OWNER
+PIN_GPIO_IO34        OWNER
+PIN_GPIO_IO35        OWNER
+PIN_GPIO_IO36        OWNER
+PIN_GPIO_IO37        OWNER
+PIN_I2C2_SCL         OWNER
+PIN_I2C2_SDA         OWNER
+PIN_PDM_BIT_STREAM0  OWNER
+PIN_PDM_CLK          OWNER
+PIN_SAI1_RXD0        OWNER
+PIN_SAI1_TXC         OWNER
+PIN_SAI1_TXD0        OWNER
+PIN_SAI1_TXFS        OWNER
+PIN_SD1_CLK          OWNER
+PIN_SD1_CMD          OWNER
+PIN_SD1_DATA0        OWNER
+PIN_SD1_DATA1        OWNER
+PIN_SD1_DATA2        OWNER
+PIN_SD1_DATA3        OWNER
+PIN_SD1_DATA4        OWNER
+PIN_SD1_DATA5        OWNER
+PIN_SD1_DATA6        OWNER
+PIN_SD1_DATA7        OWNER
+PIN_SD1_STROBE       OWNER
+PIN_SD2_CD_B         OWNER
+PIN_SD2_CLK          OWNER
+PIN_SD2_CMD          OWNER
+PIN_SD2_DATA0        OWNER
+PIN_SD2_DATA1        OWNER
+PIN_SD2_DATA2        OWNER
+PIN_SD2_DATA3        OWNER
+PIN_SD2_RESET_B      OWNER
+PIN_SD2_VSELECT      OWNER
+PIN_SD3_CLK          OWNER
+PIN_SD3_CMD          OWNER
+PIN_SD3_DATA0        OWNER
+PIN_SD3_DATA1        OWNER
+PIN_SD3_DATA2        OWNER
+PIN_SD3_DATA3        OWNER
+PIN_UART1_RXD        OWNER, test
+PIN_UART1_TXD        OWNER
+PIN_XSPI1_DATA0      OWNER
+PIN_XSPI1_DATA1      OWNER
+PIN_XSPI1_DATA2      OWNER
+PIN_XSPI1_DATA3      OWNER
+PIN_XSPI1_DATA4      OWNER
+PIN_XSPI1_DATA5      OWNER
+PIN_XSPI1_DATA6      OWNER
+PIN_XSPI1_DATA7      OWNER
+PIN_XSPI1_DQS        OWNER
+PIN_XSPI1_SCLK       OWNER
+PIN_XSPI1_SS0_B      OWNER
+PIN_XSPI1_SS1_B      OWNER
+
+# Memory
+
+OCRAM               EXEC, begin=0x020480000, size=352K
+A55_ROM             EXEC, begin=0x000000000, end=0x00003FFFF
+DDR                 EXEC, begin=0x088000000, end=0x089FFFFFF
+DDR                 EXEC, begin=0x08E000000, end=0x87FFFFFFF
+FLEXSPI1_MEM        EXEC, begin=0x000000000, end=0x0FFFFFFFF
+OCRAM_C             EXEC, begin=0x001000000, end=0x001017FFF
+GPU                 DATA, begin=0x04D900000, end=0x04DD7FFFF
+M7MIX               DATA, begin=0x02043C000, end=0x02043FFFF
+
+# RPC Interface of A55 NS agent1 (duplicated AP-NS-agent0)
+
+SCMI_AGENT3         name="AP-NS-agent1", dup=2
+MAILBOX             type=mu, mu=5, test=4
+CHANNEL             db=0, xport=smt, rpc=scmi, type=a2p
+CHANNEL             db=1, xport=smt, rpc=scmi, type=p2a_notify
+
+# RPC Interface of A55 NS agent2 (duplicated AP-NS-agent0)
+
+SCMI_AGENT4         name="AP-NS-agent2", dup=2
+MAILBOX             type=mu, mu=7, test=6
+CHANNEL             db=0, xport=smt, rpc=scmi, type=a2p
+CHANNEL             db=1, xport=smt, rpc=scmi, type=p2a_notify
-- 
2.47.0

